// @pulumi/pulumiservice/teamEnvironmentPermission.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamEnvironmentPermission = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("./utilities"));
/**
 * A permission for a team to use an environment.
 */
class TeamEnvironmentPermission extends pulumi.CustomResource {
    /**
     * Get an existing TeamEnvironmentPermission resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TeamEnvironmentPermission(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:index:TeamEnvironmentPermission';
    /**
     * Returns true if the given object is an instance of TeamEnvironmentPermission.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TeamEnvironmentPermission.__pulumiType;
    }
    /**
     * Create a TeamEnvironmentPermission resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.environment === undefined && !opts.urn) {
                throw new Error("Missing required property 'environment'");
            }
            if (args?.organization === undefined && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if (args?.permission === undefined && !opts.urn) {
                throw new Error("Missing required property 'permission'");
            }
            if (args?.team === undefined && !opts.urn) {
                throw new Error("Missing required property 'team'");
            }
            resourceInputs["environment"] = args?.environment;
            resourceInputs["maxOpenDuration"] = args?.maxOpenDuration;
            resourceInputs["organization"] = args?.organization;
            resourceInputs["permission"] = args?.permission;
            resourceInputs["project"] = (args?.project) ?? "default";
            resourceInputs["team"] = args?.team;
        }
        else {
            resourceInputs["environment"] = undefined /*out*/;
            resourceInputs["maxOpenDuration"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["permission"] = undefined /*out*/;
            resourceInputs["project"] = undefined /*out*/;
            resourceInputs["team"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["environment", "maxOpenDuration", "organization", "permission", "project", "team"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(TeamEnvironmentPermission.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TeamEnvironmentPermission = TeamEnvironmentPermission;
