// @pulumi/aws/cloudwatch/logS3TableIntegrationSource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogS3TableIntegrationSource = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a CloudWatch Logs S3 Table Integration data source association.
 * This resource associates a data source (such as a CloudWatch log type) with an S3 Table Integration, enabling CloudWatch logs to be automatically written to S3 Tables for analytics.
 *
 * For more information, see the [CloudWatch Logs S3 Tables integration documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/s3-tables-integration.html).
 *
 * ## Example Usage
 *
 * ### Associate All Sources (Wildcard)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudwatch.LogS3TableIntegrationSource("example", {
 *     integrationArn: exampleAwsObservabilityadminS3TableIntegration.arn,
 *     dataSource: {
 *         name: "*",
 *         type: "*",
 *     },
 * });
 * ```
 *
 * ### Associate a Custom Data Source
 *
 * To route log stream messages from a specific custom data source into a dedicated S3 Table, tag the CloudWatch log group with `cw:datasource:name` and `cw:datasource:type`. The integration then writes matching log streams into a table named `{name}__{type}` inside the `aws-cloudwatch` table bucket.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * // Tag the log group to declare it as a custom data source.
 * // Log stream messages are written to the "myapp__events" table
 * // in the aws-cloudwatch table bucket.
 * const example = new aws.cloudwatch.LogGroup("example", {
 *     name: "/example/myapp",
 *     tags: {
 *         "cw:datasource:name": "myapp",
 *         "cw:datasource:type": "events",
 *     },
 * });
 * const exampleLogS3TableIntegrationSource = new aws.cloudwatch.LogS3TableIntegrationSource("example", {
 *     integrationArn: exampleAwsObservabilityadminS3TableIntegration.arn,
 *     dataSource: {
 *         name: "myapp",
 *         type: "events",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `integrationArn` (String) ARN of the integration.
 * * `id` (String) ID of the association.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import S3 Table Integration data source associations using `integrationArn` and `id` separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:cloudwatch/logS3TableIntegrationSource:LogS3TableIntegrationSource example arn:aws:observabilityadmin:us-west-2:123456789012:s3tableintegration/3g5043wqe54nmw5apiugwkn1a,a8928b36-ab82-4ae2-ae5c-fcb910ec4237
 * ```
 */
class LogS3TableIntegrationSource extends pulumi.CustomResource {
    /**
     * Get an existing LogS3TableIntegrationSource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LogS3TableIntegrationSource(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:cloudwatch/logS3TableIntegrationSource:LogS3TableIntegrationSource';
    /**
     * Returns true if the given object is an instance of LogS3TableIntegrationSource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogS3TableIntegrationSource.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["dataSource"] = state?.dataSource;
            resourceInputs["integrationArn"] = state?.integrationArn;
            resourceInputs["region"] = state?.region;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.dataSource === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataSource'");
            }
            if (args?.integrationArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'integrationArn'");
            }
            resourceInputs["dataSource"] = args?.dataSource;
            resourceInputs["integrationArn"] = args?.integrationArn;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LogS3TableIntegrationSource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogS3TableIntegrationSource = LogS3TableIntegrationSource;
