// @pulumi/pulumiservice/types/input.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.deploymentSettingsVcsArgsProvideDefaults = exports.deploymentSettingsGithubArgsProvideDefaults = exports.deploymentSettingsCacheOptionsArgsProvideDefaults = void 0;
/**
 * deploymentSettingsCacheOptionsArgsProvideDefaults sets the appropriate defaults for DeploymentSettingsCacheOptionsArgs
 */
function deploymentSettingsCacheOptionsArgsProvideDefaults(val) {
    return {
        ...val,
        enable: (val.enable) ?? false,
    };
}
exports.deploymentSettingsCacheOptionsArgsProvideDefaults = deploymentSettingsCacheOptionsArgsProvideDefaults;
/**
 * deploymentSettingsGithubArgsProvideDefaults sets the appropriate defaults for DeploymentSettingsGithubArgs
 */
function deploymentSettingsGithubArgsProvideDefaults(val) {
    return {
        ...val,
        deployCommits: (val.deployCommits) ?? true,
        previewPullRequests: (val.previewPullRequests) ?? true,
        pullRequestTemplate: (val.pullRequestTemplate) ?? false,
    };
}
exports.deploymentSettingsGithubArgsProvideDefaults = deploymentSettingsGithubArgsProvideDefaults;
/**
 * deploymentSettingsVcsArgsProvideDefaults sets the appropriate defaults for DeploymentSettingsVcsArgs
 */
function deploymentSettingsVcsArgsProvideDefaults(val) {
    return {
        ...val,
        deployCommits: (val.deployCommits) ?? true,
        previewPullRequests: (val.previewPullRequests) ?? true,
        pullRequestTemplate: (val.pullRequestTemplate) ?? false,
    };
}
exports.deploymentSettingsVcsArgsProvideDefaults = deploymentSettingsVcsArgsProvideDefaults;
