// @pulumi/pulumiservice/types/output.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.insightsAccountStateProvideDefaults = exports.deploymentSettingsVcsProvideDefaults = exports.deploymentSettingsGithubProvideDefaults = exports.deploymentSettingsCacheOptionsProvideDefaults = void 0;
/**
 * deploymentSettingsCacheOptionsProvideDefaults sets the appropriate defaults for DeploymentSettingsCacheOptions
 */
function deploymentSettingsCacheOptionsProvideDefaults(val) {
    return {
        ...val,
        enable: (val.enable) ?? false,
    };
}
exports.deploymentSettingsCacheOptionsProvideDefaults = deploymentSettingsCacheOptionsProvideDefaults;
/**
 * deploymentSettingsGithubProvideDefaults sets the appropriate defaults for DeploymentSettingsGithub
 */
function deploymentSettingsGithubProvideDefaults(val) {
    return {
        ...val,
        deployCommits: (val.deployCommits) ?? true,
        previewPullRequests: (val.previewPullRequests) ?? true,
        pullRequestTemplate: (val.pullRequestTemplate) ?? false,
    };
}
exports.deploymentSettingsGithubProvideDefaults = deploymentSettingsGithubProvideDefaults;
/**
 * deploymentSettingsVcsProvideDefaults sets the appropriate defaults for DeploymentSettingsVcs
 */
function deploymentSettingsVcsProvideDefaults(val) {
    return {
        ...val,
        deployCommits: (val.deployCommits) ?? true,
        previewPullRequests: (val.previewPullRequests) ?? true,
        pullRequestTemplate: (val.pullRequestTemplate) ?? false,
    };
}
exports.deploymentSettingsVcsProvideDefaults = deploymentSettingsVcsProvideDefaults;
/**
 * insightsAccountStateProvideDefaults sets the appropriate defaults for InsightsAccountState
 */
function insightsAccountStateProvideDefaults(val) {
    return {
        ...val,
        scanSchedule: (val.scanSchedule) ?? "none",
    };
}
exports.insightsAccountStateProvideDefaults = insightsAccountStateProvideDefaults;
