// @pulumi/aws/observabilityadmin/telemetryRuleForOrganization.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TelemetryRuleForOrganization = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS CloudWatch Observability Admin Telemetry Rule for an AWS Organization.
 *
 * An organization-wide telemetry rule defines how telemetry data (logs, metrics, or traces) should be collected for AWS resources across the accounts in your organization. The rule can target one or more Regions and configure a destination (such as CloudWatch Logs or S3) along with source-specific parameters for VPC flow logs, WAF logs, CloudTrail events, ELB access logs, and more.
 *
 * > **NOTE:** Before using this resource, telemetry evaluation for organization must be enabled. Use the `aws.observabilityadmin.TelemetryEvaluationForOrganization` resource to enable it.
 *
 * > **NOTE:** This resource can only be used from the organization management account.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {});
 * const exampleTelemetryRuleForOrganization = new aws.observabilityadmin.TelemetryRuleForOrganization("example", {
 *     ruleName: "example-org-telemetry-rule",
 *     rule: {
 *         telemetryType: "Logs",
 *         resourceType: "AWS::EC2::VPC",
 *     },
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ### VPC Flow Logs to CloudWatch Logs
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {});
 * const exampleTelemetryRuleForOrganization = new aws.observabilityadmin.TelemetryRuleForOrganization("example", {
 *     ruleName: "org-vpc-flow-logs-rule",
 *     rule: {
 *         telemetryType: "Logs",
 *         resourceType: "AWS::EC2::VPC",
 *         telemetrySourceTypes: ["VPC_FLOW_LOGS"],
 *         allRegions: true,
 *         allowFieldUpdates: true,
 *         destinationConfiguration: {
 *             destinationType: "cloud-watch-logs",
 *             destinationPattern: "/aws/vpcflowlogs/<resourceId>",
 *             retentionInDays: 30,
 *             vpcFlowLogParameters: {
 *                 trafficType: "ALL",
 *                 maxAggregationInterval: 60,
 *             },
 *         },
 *     },
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ### Scoped to Specific Organizational Units
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const current = aws.organizations.getOrganization({});
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {});
 * const exampleTelemetryRuleForOrganization = new aws.observabilityadmin.TelemetryRuleForOrganization("example", {
 *     ruleName: "org-scoped-rule",
 *     rule: {
 *         telemetryType: "Logs",
 *         resourceType: "AWS::EKS::Cluster",
 *         scope: current.then(current => `OrganizationId = '${current.id}'`),
 *         selectionCriteria: "ResourceTags.Environment = 'production'",
 *         regions: [
 *             "us-east-1",
 *             "us-west-2",
 *         ],
 *     },
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ### With Tags
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.observabilityadmin.TelemetryEvaluationForOrganization("example", {});
 * const exampleTelemetryRuleForOrganization = new aws.observabilityadmin.TelemetryRuleForOrganization("example", {
 *     ruleName: "org-tagged-rule",
 *     rule: {
 *         telemetryType: "Logs",
 *         resourceType: "AWS::EC2::VPC",
 *     },
 *     tags: {
 *         Environment: "production",
 *         Purpose: "organization-monitoring",
 *     },
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `ruleName` (String) Name of the telemetry rule.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import CloudWatch Observability Admin Telemetry Rules for Organization using `ruleName`. For example:
 *
 * ```sh
 * $ pulumi import aws:observabilityadmin/telemetryRuleForOrganization:TelemetryRuleForOrganization example example-org-telemetry-rule
 * ```
 */
class TelemetryRuleForOrganization extends pulumi.CustomResource {
    /**
     * Get an existing TelemetryRuleForOrganization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TelemetryRuleForOrganization(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:observabilityadmin/telemetryRuleForOrganization:TelemetryRuleForOrganization';
    /**
     * Returns true if the given object is an instance of TelemetryRuleForOrganization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TelemetryRuleForOrganization.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["rule"] = state?.rule;
            resourceInputs["ruleArn"] = state?.ruleArn;
            resourceInputs["ruleName"] = state?.ruleName;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.rule === undefined && !opts.urn) {
                throw new Error("Missing required property 'rule'");
            }
            if (args?.ruleName === undefined && !opts.urn) {
                throw new Error("Missing required property 'ruleName'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["rule"] = args?.rule;
            resourceInputs["ruleName"] = args?.ruleName;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["ruleArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TelemetryRuleForOrganization.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TelemetryRuleForOrganization = TelemetryRuleForOrganization;
