// @pulumi/aws/appautoscaling/policy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an Application AutoScaling Policy resource.
 *
 * ## Example Usage
 *
 * ### DynamoDB Table Autoscaling
 *
 * ### ECS Service Autoscaling
 *
 * ### Preserve desired count when updating an autoscaled ECS Service
 *
 * > To preserve the `desiredCount` of an autoscaled ECS Service during updates, use the [`ignoreChanges` resource option](https://www.pulumi.com/docs/concepts/options/ignorechanges/). This prevents Pulumi from resetting the count that autoscaling has set:
 * > ```typescript
 * > const ecsService = new aws.ecs.Service("ecsService", {
 * >     name: "serviceName",
 * >     cluster: "clusterName",
 * >     taskDefinition: "taskDefinitionFamily:1",
 * >     desiredCount: 2,
 * > }, {
 * >     ignoreChanges: ["desiredCount"],
 * > });
 * > ```
 *
 * ### Aurora Read Replica Autoscaling
 *
 * ### Create target tracking scaling policy using metric math
 *
 * ### Predictive Scaling
 *
 * ### MSK / Kafka Autoscaling
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `name` - (String) Name of the scaling policy.
 * * `resourceId` - (String) Resource type and unique identifier string for the resource associated with the scaling policy.
 * * `scalableDimension` - (String) Scalable dimension of the scalable target.
 * * `serviceNamespace` - (String) AWS service namespace of the scalable target.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Application AutoScaling Policy using the `service-namespace` , `resource-id`, `scalable-dimension` and `policy-name` separated by `/`. For example:
 *
 * ```sh
 * $ pulumi import aws:appautoscaling/policy:Policy test-policy service-namespace/resource-id/scalable-dimension/policy-name
 * ```
 */
class Policy extends pulumi.CustomResource {
    /**
     * Get an existing Policy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Policy(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:appautoscaling/policy:Policy';
    /**
     * Returns true if the given object is an instance of Policy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Policy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["alarmArns"] = state?.alarmArns;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["name"] = state?.name;
            resourceInputs["policyType"] = state?.policyType;
            resourceInputs["predictiveScalingPolicyConfiguration"] = state?.predictiveScalingPolicyConfiguration;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceId"] = state?.resourceId;
            resourceInputs["scalableDimension"] = state?.scalableDimension;
            resourceInputs["serviceNamespace"] = state?.serviceNamespace;
            resourceInputs["stepScalingPolicyConfiguration"] = state?.stepScalingPolicyConfiguration;
            resourceInputs["targetTrackingScalingPolicyConfiguration"] = state?.targetTrackingScalingPolicyConfiguration;
        }
        else {
            const args = argsOrState;
            if (args?.resourceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceId'");
            }
            if (args?.scalableDimension === undefined && !opts.urn) {
                throw new Error("Missing required property 'scalableDimension'");
            }
            if (args?.serviceNamespace === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceNamespace'");
            }
            resourceInputs["name"] = args?.name;
            resourceInputs["policyType"] = args?.policyType;
            resourceInputs["predictiveScalingPolicyConfiguration"] = args?.predictiveScalingPolicyConfiguration;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceId"] = args?.resourceId;
            resourceInputs["scalableDimension"] = args?.scalableDimension;
            resourceInputs["serviceNamespace"] = args?.serviceNamespace;
            resourceInputs["stepScalingPolicyConfiguration"] = args?.stepScalingPolicyConfiguration;
            resourceInputs["targetTrackingScalingPolicyConfiguration"] = args?.targetTrackingScalingPolicyConfiguration;
            resourceInputs["alarmArns"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Policy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Policy = Policy;
