// @pulumi/pulumiservice/api/auth/saml.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAML = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
/**
 * Updates the SAML configuration for a SAML-backed organization, including the identity provider SSO descriptor, attribute mappings, and other SAML settings. The new IDP SSO descriptor is required in the update request.
 */
class SAML extends pulumi.CustomResource {
    /**
     * Get an existing SAML resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SAML(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:api/auth:SAML';
    /**
     * Returns true if the given object is an instance of SAML.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SAML.__pulumiType;
    }
    /**
     * Create a SAML resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.orgName === undefined && !opts.urn) {
                throw new Error("Missing required property 'orgName'");
            }
            resourceInputs["newIdpSsoDescriptor"] = args?.newIdpSsoDescriptor;
            resourceInputs["orgName"] = args?.orgName;
            resourceInputs["entityId"] = undefined /*out*/;
            resourceInputs["idpSsoDescriptor"] = undefined /*out*/;
            resourceInputs["nameIdFormat"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["ssoUrl"] = undefined /*out*/;
            resourceInputs["validUntil"] = undefined /*out*/;
            resourceInputs["validationError"] = undefined /*out*/;
        }
        else {
            resourceInputs["entityId"] = undefined /*out*/;
            resourceInputs["idpSsoDescriptor"] = undefined /*out*/;
            resourceInputs["nameIdFormat"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["ssoUrl"] = undefined /*out*/;
            resourceInputs["validUntil"] = undefined /*out*/;
            resourceInputs["validationError"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SAML.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SAML = SAML;
