// @pulumi/aws/pinpoint/smsvoicev2EventDestination.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Smsvoicev2EventDestination = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS End User Messaging SMS Event Destination.
 *
 * An event destination is a location where messaging events are published. Exactly one of `cloudwatchLogsDestination`, `kinesisFirehoseDestination`, or `snsDestination` must be configured per event destination. Changing the sink type (e.g., from `snsDestination` to `cloudwatchLogsDestination`) forces resource replacement — AWS's `UpdateEventDestination` rejects sink-type changes with `ConflictException`.
 *
 * ## Example Usage
 *
 * ### CloudWatch Logs Destination
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.pinpoint.Smsvoicev2ConfigurationSet("example", {name: "example-configuration-set"});
 * const exampleSmsvoicev2EventDestination = new aws.pinpoint.Smsvoicev2EventDestination("example", {
 *     configurationSetName: example.name,
 *     eventDestinationName: "example",
 *     matchingEventTypes: ["ALL"],
 *     cloudwatchLogsDestination: {
 *         iamRoleArn: exampleAwsIamRole.arn,
 *         logGroupArn: exampleAwsCloudwatchLogGroup.arn,
 *     },
 * });
 * ```
 *
 * ### Kinesis Firehose Destination
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.pinpoint.Smsvoicev2ConfigurationSet("example", {name: "example-configuration-set"});
 * const exampleSmsvoicev2EventDestination = new aws.pinpoint.Smsvoicev2EventDestination("example", {
 *     configurationSetName: example.name,
 *     eventDestinationName: "example",
 *     matchingEventTypes: ["ALL"],
 *     kinesisFirehoseDestination: {
 *         deliveryStreamArn: exampleAwsKinesisFirehoseDeliveryStream.arn,
 *         iamRoleArn: exampleAwsIamRole.arn,
 *     },
 * });
 * ```
 *
 * ### SNS Destination
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.pinpoint.Smsvoicev2ConfigurationSet("example", {name: "example-configuration-set"});
 * const exampleSmsvoicev2EventDestination = new aws.pinpoint.Smsvoicev2EventDestination("example", {
 *     configurationSetName: example.name,
 *     eventDestinationName: "example",
 *     matchingEventTypes: ["ALL"],
 *     snsDestination: {
 *         topicArn: exampleAwsSnsTopic.arn,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `configurationSetName` (String) Name of the configuration set this event destination belongs to.
 * * `eventDestinationName` (String) Name of the event destination.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import an event destination using the `configurationSetName` and `eventDestinationName`, separated by a comma. For example:
 *
 * ```sh
 * $ pulumi import aws:pinpoint/smsvoicev2EventDestination:Smsvoicev2EventDestination example "example-configuration-set,example-event-destination"
 * ```
 */
class Smsvoicev2EventDestination extends pulumi.CustomResource {
    /**
     * Get an existing Smsvoicev2EventDestination resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Smsvoicev2EventDestination(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:pinpoint/smsvoicev2EventDestination:Smsvoicev2EventDestination';
    /**
     * Returns true if the given object is an instance of Smsvoicev2EventDestination.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Smsvoicev2EventDestination.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cloudwatchLogsDestination"] = state?.cloudwatchLogsDestination;
            resourceInputs["configurationSetArn"] = state?.configurationSetArn;
            resourceInputs["configurationSetName"] = state?.configurationSetName;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["eventDestinationName"] = state?.eventDestinationName;
            resourceInputs["kinesisFirehoseDestination"] = state?.kinesisFirehoseDestination;
            resourceInputs["matchingEventTypes"] = state?.matchingEventTypes;
            resourceInputs["region"] = state?.region;
            resourceInputs["snsDestination"] = state?.snsDestination;
        }
        else {
            const args = argsOrState;
            if (args?.configurationSetName === undefined && !opts.urn) {
                throw new Error("Missing required property 'configurationSetName'");
            }
            if (args?.eventDestinationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'eventDestinationName'");
            }
            if (args?.matchingEventTypes === undefined && !opts.urn) {
                throw new Error("Missing required property 'matchingEventTypes'");
            }
            resourceInputs["cloudwatchLogsDestination"] = args?.cloudwatchLogsDestination;
            resourceInputs["configurationSetName"] = args?.configurationSetName;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["eventDestinationName"] = args?.eventDestinationName;
            resourceInputs["kinesisFirehoseDestination"] = args?.kinesisFirehoseDestination;
            resourceInputs["matchingEventTypes"] = args?.matchingEventTypes;
            resourceInputs["region"] = args?.region;
            resourceInputs["snsDestination"] = args?.snsDestination;
            resourceInputs["configurationSetArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Smsvoicev2EventDestination.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Smsvoicev2EventDestination = Smsvoicev2EventDestination;
