// @pulumi/pulumiservice/api/orgTemplateCollection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgTemplateCollection = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Creates a new template collection (source) for an organization. Template collections define where project templates are sourced from, such as a Git repository. Organization members can use these templates to create new stacks with pre-configured infrastructure code.
 */
class OrgTemplateCollection extends pulumi.CustomResource {
    /**
     * Get an existing OrgTemplateCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OrgTemplateCollection(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:api:OrgTemplateCollection';
    /**
     * Returns true if the given object is an instance of OrgTemplateCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OrgTemplateCollection.__pulumiType;
    }
    /**
     * Create a OrgTemplateCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.orgName === undefined && !opts.urn) {
                throw new Error("Missing required property 'orgName'");
            }
            if (args?.sourceURL === undefined && !opts.urn) {
                throw new Error("Missing required property 'sourceURL'");
            }
            resourceInputs["destination"] = args?.destination;
            resourceInputs["destinationURL"] = args?.destinationURL;
            resourceInputs["name"] = args?.name;
            resourceInputs["orgName"] = args?.orgName;
            resourceInputs["sourceURL"] = args?.sourceURL;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["isValid"] = undefined /*out*/;
            resourceInputs["templateID"] = undefined /*out*/;
        }
        else {
            resourceInputs["destination"] = undefined /*out*/;
            resourceInputs["destinationURL"] = undefined /*out*/;
            resourceInputs["error"] = undefined /*out*/;
            resourceInputs["isValid"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["sourceURL"] = undefined /*out*/;
            resourceInputs["templateID"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(OrgTemplateCollection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OrgTemplateCollection = OrgTemplateCollection;
