// @pulumi/aws/s3control/multiRegionAccessPointRoutes.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiRegionAccessPointRoutes = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides a resource to manage the routing configuration for an S3 Multi-Region Access Point.
 *
 * > Note: Destruction of this resource only removes it from state. It __does not__ alter the configured traffic routing percentages.
 *
 * ## Example Usage
 *
 * ### Active-Active Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const primary = new aws.s3.Bucket("primary", {bucket: "example-bucket-primary"});
 * const secondary = new aws.s3.Bucket("secondary", {bucket: "example-bucket-secondary"});
 * const example = new aws.s3control.MultiRegionAccessPoint("example", {details: {
 *     name: "example",
 *     regions: [
 *         {
 *             bucket: primary.bucket,
 *         },
 *         {
 *             bucket: secondary.bucket,
 *         },
 *     ],
 * }});
 * const exampleMultiRegionAccessPointRoutes = new aws.s3control.MultiRegionAccessPointRoutes("example", {
 *     mrap: example.arn,
 *     routes: [
 *         {
 *             bucket: primary.bucket,
 *             region: primary.bucketRegion,
 *             trafficDialPercentage: 100,
 *         },
 *         {
 *             bucket: secondary.bucket,
 *             region: secondary.bucketRegion,
 *             trafficDialPercentage: 100,
 *         },
 *     ],
 * });
 * ```
 *
 * ### Failover Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.s3control.MultiRegionAccessPointRoutes("example", {
 *     mrap: exampleAwsS3controlMultiRegionAccessPoint.arn,
 *     routes: [
 *         {
 *             bucket: primary.bucket,
 *             region: primary.bucketRegion,
 *             trafficDialPercentage: 0,
 *         },
 *         {
 *             bucket: secondary.bucket,
 *             region: secondary.bucketRegion,
 *             trafficDialPercentage: 100,
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `mrap` (String) ARN of the Multi-Region Access Point.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Multi-Region Access Point Routes using the `mrap` argument. For example:
 *
 * ```sh
 * $ pulumi import aws:s3control/multiRegionAccessPointRoutes:MultiRegionAccessPointRoutes example arn:aws:s3::0123456789012:accesspoint/example
 * ```
 */
class MultiRegionAccessPointRoutes extends pulumi.CustomResource {
    /**
     * Get an existing MultiRegionAccessPointRoutes resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MultiRegionAccessPointRoutes(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:s3control/multiRegionAccessPointRoutes:MultiRegionAccessPointRoutes';
    /**
     * Returns true if the given object is an instance of MultiRegionAccessPointRoutes.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MultiRegionAccessPointRoutes.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["mrap"] = state?.mrap;
            resourceInputs["region"] = state?.region;
            resourceInputs["routes"] = state?.routes;
        }
        else {
            const args = argsOrState;
            if (args?.mrap === undefined && !opts.urn) {
                throw new Error("Missing required property 'mrap'");
            }
            if (args?.routes === undefined && !opts.urn) {
                throw new Error("Missing required property 'routes'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["mrap"] = args?.mrap;
            resourceInputs["region"] = args?.region;
            resourceInputs["routes"] = args?.routes;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MultiRegionAccessPointRoutes.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MultiRegionAccessPointRoutes = MultiRegionAccessPointRoutes;
