// @pulumi/pulumiservice/deploymentSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentSettings = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const inputs = __importStar(require("./types/input"));
const utilities = __importStar(require("./utilities"));
/**
 * Deployment settings configure Pulumi Deployments for a stack.
 *
 * ### Import
 *
 * Deployment settings can be imported using the `id`, which for deployment settings is `{org}/{project}/{stack}` e.g.,
 *
 * ```sh
 *  $ pulumi import pulumiservice:index:DeploymentSettings my_settings my-org/my-project/my-stack
 * ```
 */
class DeploymentSettings extends pulumi.CustomResource {
    /**
     * Get an existing DeploymentSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentSettings(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:index:DeploymentSettings';
    /**
     * Returns true if the given object is an instance of DeploymentSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentSettings.__pulumiType;
    }
    /**
     * Create a DeploymentSettings resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.organization === undefined && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if (args?.project === undefined && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if (args?.stack === undefined && !opts.urn) {
                throw new Error("Missing required property 'stack'");
            }
            resourceInputs["agentPoolId"] = args?.agentPoolId;
            resourceInputs["cacheOptions"] = args ? pulumi.output(args.cacheOptions).apply(v => v === undefined ? undefined : inputs.deploymentSettingsCacheOptionsArgsProvideDefaults(v)) : undefined;
            resourceInputs["executorContext"] = args?.executorContext;
            resourceInputs["github"] = args ? pulumi.output(args.github).apply(v => v === undefined ? undefined : inputs.deploymentSettingsGithubArgsProvideDefaults(v)) : undefined;
            resourceInputs["operationContext"] = args?.operationContext;
            resourceInputs["organization"] = args?.organization;
            resourceInputs["project"] = args?.project;
            resourceInputs["sourceContext"] = args?.sourceContext;
            resourceInputs["stack"] = args?.stack;
            resourceInputs["vcs"] = args ? pulumi.output(args.vcs).apply(v => v === undefined ? undefined : inputs.deploymentSettingsVcsArgsProvideDefaults(v)) : undefined;
        }
        else {
            resourceInputs["agentPoolId"] = undefined /*out*/;
            resourceInputs["cacheOptions"] = undefined /*out*/;
            resourceInputs["executorContext"] = undefined /*out*/;
            resourceInputs["github"] = undefined /*out*/;
            resourceInputs["operationContext"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["project"] = undefined /*out*/;
            resourceInputs["sourceContext"] = undefined /*out*/;
            resourceInputs["stack"] = undefined /*out*/;
            resourceInputs["vcs"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeploymentSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeploymentSettings = DeploymentSettings;
