// @pulumi/pulumiservice/insightsAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsightsAccount = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("./utilities"));
/**
 * Insights Account for cloud resource scanning and analysis across AWS, Azure, and GCP.
 */
class InsightsAccount extends pulumi.CustomResource {
    /**
     * Get an existing InsightsAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InsightsAccount(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:index:InsightsAccount';
    /**
     * Returns true if the given object is an instance of InsightsAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InsightsAccount.__pulumiType;
    }
    /**
     * Create a InsightsAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.accountName === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if (args?.environment === undefined && !opts.urn) {
                throw new Error("Missing required property 'environment'");
            }
            if (args?.organizationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'organizationName'");
            }
            if (args?.provider === undefined && !opts.urn) {
                throw new Error("Missing required property 'provider'");
            }
            if (args?.scanSchedule === undefined && !opts.urn) {
                throw new Error("Missing required property 'scanSchedule'");
            }
            resourceInputs["accountName"] = args?.accountName;
            resourceInputs["environment"] = args?.environment;
            resourceInputs["organizationName"] = args?.organizationName;
            resourceInputs["provider"] = args?.provider;
            resourceInputs["providerConfig"] = args?.providerConfig;
            resourceInputs["scanSchedule"] = (args?.scanSchedule) ?? "none";
            resourceInputs["tags"] = args?.tags;
            resourceInputs["insightsAccountId"] = undefined /*out*/;
            resourceInputs["scheduledScanEnabled"] = undefined /*out*/;
        }
        else {
            resourceInputs["accountName"] = undefined /*out*/;
            resourceInputs["environment"] = undefined /*out*/;
            resourceInputs["insightsAccountId"] = undefined /*out*/;
            resourceInputs["organizationName"] = undefined /*out*/;
            resourceInputs["provider"] = undefined /*out*/;
            resourceInputs["providerConfig"] = undefined /*out*/;
            resourceInputs["scanSchedule"] = undefined /*out*/;
            resourceInputs["scheduledScanEnabled"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["accountName", "organizationName", "provider"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(InsightsAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InsightsAccount = InsightsAccount;
