// @pulumi/aws/ecs/daemon.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Daemon = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Provides an ECS Daemon resource, which manages a daemon that runs exactly one task on each container instance in an ECS cluster.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.Daemon("example", {
 *     name: "example-daemon",
 *     clusterArn: exampleAwsEcsCluster.arn,
 *     daemonTaskDefinitionArn: exampleAwsEcsDaemonTaskDefinition.arn,
 *     capacityProviderArns: [exampleAwsEcsCapacityProvider.arn],
 * });
 * ```
 *
 * ### With Deployment Configuration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.Daemon("example", {
 *     name: "example-daemon",
 *     clusterArn: exampleAwsEcsCluster.arn,
 *     daemonTaskDefinitionArn: exampleAwsEcsDaemonTaskDefinition.arn,
 *     capacityProviderArns: [exampleAwsEcsCapacityProvider.arn],
 *     deploymentConfiguration: {
 *         drainPercent: 50,
 *         bakeTimeInMinutes: 10,
 *         alarms: {
 *             alarmNames: ["example-alarm"],
 *             enable: true,
 *         },
 *     },
 * });
 * ```
 *
 * ### With Tags
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.Daemon("example", {
 *     name: "example-daemon",
 *     clusterArn: exampleAwsEcsCluster.arn,
 *     daemonTaskDefinitionArn: exampleAwsEcsDaemonTaskDefinition.arn,
 *     capacityProviderArns: [exampleAwsEcsCapacityProvider.arn],
 *     tags: {
 *         Environment: "production",
 *         Application: "example",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `arn` (String) ARN of the ECS Daemon.
 *
 * Using `pulumi import`, import ECS Daemons using the ARN. For example:
 *
 * ```sh
 * $ pulumi import aws:ecs/daemon:Daemon example arn:aws:ecs:us-east-1:123456789012:daemon/example-cluster/example-daemon
 * ```
 */
class Daemon extends pulumi.CustomResource {
    /**
     * Get an existing Daemon resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Daemon(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ecs/daemon:Daemon';
    /**
     * Returns true if the given object is an instance of Daemon.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Daemon.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["capacityProviderArns"] = state?.capacityProviderArns;
            resourceInputs["clusterArn"] = state?.clusterArn;
            resourceInputs["daemonTaskDefinitionArn"] = state?.daemonTaskDefinitionArn;
            resourceInputs["deploymentArn"] = state?.deploymentArn;
            resourceInputs["deploymentConfiguration"] = state?.deploymentConfiguration;
            resourceInputs["enableEcsManagedTags"] = state?.enableEcsManagedTags;
            resourceInputs["enableExecuteCommand"] = state?.enableExecuteCommand;
            resourceInputs["name"] = state?.name;
            resourceInputs["propagateTags"] = state?.propagateTags;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.capacityProviderArns === undefined && !opts.urn) {
                throw new Error("Missing required property 'capacityProviderArns'");
            }
            if (args?.daemonTaskDefinitionArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'daemonTaskDefinitionArn'");
            }
            resourceInputs["capacityProviderArns"] = args?.capacityProviderArns;
            resourceInputs["clusterArn"] = args?.clusterArn;
            resourceInputs["daemonTaskDefinitionArn"] = args?.daemonTaskDefinitionArn;
            resourceInputs["deploymentConfiguration"] = args?.deploymentConfiguration;
            resourceInputs["enableEcsManagedTags"] = args?.enableEcsManagedTags;
            resourceInputs["enableExecuteCommand"] = args?.enableExecuteCommand;
            resourceInputs["name"] = args?.name;
            resourceInputs["propagateTags"] = args?.propagateTags;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["deploymentArn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Daemon.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Daemon = Daemon;
