// @pulumi/aws/ecs/daemonTaskDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaemonTaskDefinition = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages a revision of an ECS daemon task definition for use with daemon scheduling strategy.
 *
 * ## Example Usage
 *
 * ### Basic Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.DaemonTaskDefinition("example", {
 *     family: "my-daemon-service",
 *     cpu: "512",
 *     memory: "1024",
 *     containerDefinitions: [{
 *         name: "app",
 *         image: "nginx:latest",
 *         cpu: 256,
 *         memory: 512,
 *         essential: true,
 *     }],
 * });
 * ```
 *
 * ### With IAM Roles
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const taskExecution = new aws.iam.Role("task_execution", {
 *     name: "daemon-task-execution-role",
 *     assumeRolePolicy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Action: "sts:AssumeRole",
 *             Effect: "Allow",
 *             Principal: {
 *                 Service: "ecs-tasks.amazonaws.com",
 *             },
 *         }],
 *     }),
 * });
 * const task = new aws.iam.Role("task", {
 *     name: "daemon-task-role",
 *     assumeRolePolicy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Action: "sts:AssumeRole",
 *             Effect: "Allow",
 *             Principal: {
 *                 Service: "ecs-tasks.amazonaws.com",
 *             },
 *         }],
 *     }),
 * });
 * const example = new aws.ecs.DaemonTaskDefinition("example", {
 *     family: "my-daemon-service",
 *     executionRoleArn: taskExecution.arn,
 *     taskRoleArn: task.arn,
 *     cpu: "512",
 *     memory: "1024",
 *     containerDefinitions: [{
 *         name: "app",
 *         image: "nginx:latest",
 *         cpu: 256,
 *         memory: 512,
 *         essential: true,
 *     }],
 * });
 * ```
 *
 * ### With Volumes
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.DaemonTaskDefinition("example", {
 *     family: "my-daemon-service",
 *     cpu: "512",
 *     memory: "1024",
 *     containerDefinitions: [{
 *         name: "app",
 *         image: "nginx:latest",
 *         cpu: 256,
 *         memory: 512,
 *         essential: true,
 *     }],
 *     volumes: [
 *         {
 *             name: "data-volume",
 *             hosts: [{
 *                 sourcePath: "/data",
 *             }],
 *         },
 *         {
 *             name: "logs-volume",
 *             hosts: [{
 *                 sourcePath: "/var/log",
 *             }],
 *         },
 *     ],
 * });
 * ```
 *
 * ### With Multiple Containers
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ecs.DaemonTaskDefinition("example", {
 *     family: "my-daemon-service",
 *     cpu: "512",
 *     memory: "1024",
 *     containerDefinitions: [
 *         {
 *             name: "app",
 *             image: "my-app:latest",
 *             cpu: 256,
 *             memory: 512,
 *             essential: true,
 *         },
 *         {
 *             name: "sidecar",
 *             image: "datadog/agent:latest",
 *             cpu: 128,
 *             memory: 256,
 *             essential: false,
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `arn` (String) ARN of the ECS Daemon Task Definition.
 *
 * Using `pulumi import`, import ECS Daemon Task Definitions using their ARNs. For example:
 *
 * ```sh
 * $ pulumi import aws:ecs/daemonTaskDefinition:DaemonTaskDefinition example arn:aws:ecs:us-east-1:012345678910:daemon-task-definition/mydaemonfamily:123
 * ```
 */
class DaemonTaskDefinition extends pulumi.CustomResource {
    /**
     * Get an existing DaemonTaskDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DaemonTaskDefinition(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:ecs/daemonTaskDefinition:DaemonTaskDefinition';
    /**
     * Returns true if the given object is an instance of DaemonTaskDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DaemonTaskDefinition.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["containerDefinitions"] = state?.containerDefinitions;
            resourceInputs["cpu"] = state?.cpu;
            resourceInputs["executionRoleArn"] = state?.executionRoleArn;
            resourceInputs["family"] = state?.family;
            resourceInputs["memory"] = state?.memory;
            resourceInputs["region"] = state?.region;
            resourceInputs["revision"] = state?.revision;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["taskRoleArn"] = state?.taskRoleArn;
            resourceInputs["volumes"] = state?.volumes;
        }
        else {
            const args = argsOrState;
            if (args?.containerDefinitions === undefined && !opts.urn) {
                throw new Error("Missing required property 'containerDefinitions'");
            }
            if (args?.family === undefined && !opts.urn) {
                throw new Error("Missing required property 'family'");
            }
            resourceInputs["containerDefinitions"] = args?.containerDefinitions;
            resourceInputs["cpu"] = args?.cpu;
            resourceInputs["executionRoleArn"] = args?.executionRoleArn;
            resourceInputs["family"] = args?.family;
            resourceInputs["memory"] = args?.memory;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["taskRoleArn"] = args?.taskRoleArn;
            resourceInputs["volumes"] = args?.volumes;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["revision"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DaemonTaskDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DaemonTaskDefinition = DaemonTaskDefinition;
