// @pulumi/aws/bedrock/agentcoreOnlineEvaluationConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentcoreOnlineEvaluationConfig = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS Bedrock AgentCore Online Evaluation Configuration. Online evaluation configurations continuously monitor agent performance by sampling live traffic from CloudWatch logs and applying evaluators to assess agent quality in production.
 *
 * > **Note:** CloudWatch Transaction Serach must be [enabled](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Enable-TransactionSearch.html) before using this resource.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.iam.Role("example", {
 *     name: "agentcore-eval-role",
 *     assumeRolePolicy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Effect: "Allow",
 *             Action: "sts:AssumeRole",
 *             Principal: {
 *                 Service: "bedrock-agentcore.amazonaws.com",
 *             },
 *         }],
 *     }),
 * });
 * const exampleLogGroup = new aws.cloudwatch.LogGroup("example", {name: "/aws/agentcore/my-agent-traces"});
 * const exampleAgentcoreOnlineEvaluationConfig = new aws.bedrock.AgentcoreOnlineEvaluationConfig("example", {
 *     onlineEvaluationConfigName: "my_evaluation_config",
 *     description: "Continuous evaluation of agent performance",
 *     enableOnCreate: true,
 *     evaluationExecutionRoleArn: example.arn,
 *     dataSourceConfig: {
 *         cloudwatchLogs: {
 *             logGroupNames: [exampleLogGroup.name],
 *             serviceNames: ["my_agent_service"],
 *         },
 *     },
 *     evaluators: [
 *         {
 *             evaluatorId: "Builtin.Helpfulness",
 *         },
 *         {
 *             evaluatorId: "Builtin.GoalSuccessRate",
 *         },
 *     ],
 *     rule: {
 *         samplingConfig: {
 *             samplingPercentage: 10,
 *         },
 *     },
 * });
 * ```
 *
 * ### With Filters and Session Config
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const filtered = new aws.bedrock.AgentcoreOnlineEvaluationConfig("filtered", {
 *     onlineEvaluationConfigName: "filtered_evaluation",
 *     enableOnCreate: true,
 *     evaluationExecutionRoleArn: exampleAwsIamRole.arn,
 *     dataSourceConfig: {
 *         cloudwatchLogs: {
 *             logGroupNames: [example.name],
 *             serviceNames: ["my_agent_service"],
 *         },
 *     },
 *     evaluators: [{
 *         evaluatorId: "Builtin.Helpfulness",
 *     }],
 *     rule: {
 *         samplingConfig: {
 *             samplingPercentage: 50,
 *         },
 *         filters: [{
 *             key: "environment",
 *             operator: "Equals",
 *             value: {
 *                 stringValue: "production",
 *             },
 *         }],
 *         sessionConfig: {
 *             sessionTimeoutMinutes: 30,
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `onlineEvaluationConfigId` (String) ID of the online evaluation config.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Bedrock AgentCore Online Evaluation Configs using `onlineEvaluationConfigId`. For example:
 *
 * ```sh
 * $ pulumi import aws:bedrock/agentcoreOnlineEvaluationConfig:AgentcoreOnlineEvaluationConfig example my_evaluation_config-aBcDeFgHiJ
 * ```
 */
class AgentcoreOnlineEvaluationConfig extends pulumi.CustomResource {
    /**
     * Get an existing AgentcoreOnlineEvaluationConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AgentcoreOnlineEvaluationConfig(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:bedrock/agentcoreOnlineEvaluationConfig:AgentcoreOnlineEvaluationConfig';
    /**
     * Returns true if the given object is an instance of AgentcoreOnlineEvaluationConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AgentcoreOnlineEvaluationConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["dataSourceConfig"] = state?.dataSourceConfig;
            resourceInputs["description"] = state?.description;
            resourceInputs["enableOnCreate"] = state?.enableOnCreate;
            resourceInputs["evaluationExecutionRoleArn"] = state?.evaluationExecutionRoleArn;
            resourceInputs["evaluators"] = state?.evaluators;
            resourceInputs["executionStatus"] = state?.executionStatus;
            resourceInputs["onlineEvaluationConfigArn"] = state?.onlineEvaluationConfigArn;
            resourceInputs["onlineEvaluationConfigId"] = state?.onlineEvaluationConfigId;
            resourceInputs["onlineEvaluationConfigName"] = state?.onlineEvaluationConfigName;
            resourceInputs["outputConfigs"] = state?.outputConfigs;
            resourceInputs["region"] = state?.region;
            resourceInputs["rule"] = state?.rule;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.dataSourceConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'dataSourceConfig'");
            }
            if (args?.enableOnCreate === undefined && !opts.urn) {
                throw new Error("Missing required property 'enableOnCreate'");
            }
            if (args?.evaluationExecutionRoleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'evaluationExecutionRoleArn'");
            }
            if (args?.evaluators === undefined && !opts.urn) {
                throw new Error("Missing required property 'evaluators'");
            }
            if (args?.onlineEvaluationConfigName === undefined && !opts.urn) {
                throw new Error("Missing required property 'onlineEvaluationConfigName'");
            }
            if (args?.rule === undefined && !opts.urn) {
                throw new Error("Missing required property 'rule'");
            }
            resourceInputs["dataSourceConfig"] = args?.dataSourceConfig;
            resourceInputs["description"] = args?.description;
            resourceInputs["enableOnCreate"] = args?.enableOnCreate;
            resourceInputs["evaluationExecutionRoleArn"] = args?.evaluationExecutionRoleArn;
            resourceInputs["evaluators"] = args?.evaluators;
            resourceInputs["executionStatus"] = args?.executionStatus;
            resourceInputs["onlineEvaluationConfigName"] = args?.onlineEvaluationConfigName;
            resourceInputs["region"] = args?.region;
            resourceInputs["rule"] = args?.rule;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["onlineEvaluationConfigArn"] = undefined /*out*/;
            resourceInputs["onlineEvaluationConfigId"] = undefined /*out*/;
            resourceInputs["outputConfigs"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AgentcoreOnlineEvaluationConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AgentcoreOnlineEvaluationConfig = AgentcoreOnlineEvaluationConfig;
