// @pulumi/aws/bedrock/agentcoreHarness.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentcoreHarness = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS Bedrock AgentCore Harness. A Harness is a managed agent loop that wraps model configuration, tools, skills, memory, and compute environment into a single deployable unit.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const assumeRole = aws.iam.getPolicyDocument({
 *     statements: [{
 *         effect: "Allow",
 *         actions: ["sts:AssumeRole"],
 *         principals: [{
 *             type: "Service",
 *             identifiers: ["bedrock-agentcore.amazonaws.com"],
 *         }],
 *     }],
 * });
 * const example = new aws.iam.Role("example", {
 *     name: "bedrock-agentcore-harness-role",
 *     assumeRolePolicy: assumeRole.then(assumeRole => assumeRole.json),
 * });
 * const exampleRolePolicy = new aws.iam.RolePolicy("example", {
 *     role: example.name,
 *     policy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Effect: "Allow",
 *             Action: [
 *                 "bedrock:InvokeModel",
 *                 "bedrock:InvokeModelWithResponseStream",
 *             ],
 *             Resource: "*",
 *         }],
 *     }),
 * });
 * const exampleAgentcoreHarness = new aws.bedrock.AgentcoreHarness("example", {
 *     harnessName: "example_harness",
 *     executionRoleArn: example.arn,
 *     model: {
 *         bedrockModelConfig: {
 *             modelId: "anthropic.claude-sonnet-4-20250514",
 *         },
 *     },
 *     systemPrompts: [{
 *         text: "You are a helpful assistant.",
 *     }],
 * });
 * ```
 *
 * ### With Tools and Truncation
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.bedrock.AgentcoreHarness("example", {
 *     harnessName: "example_with_tools",
 *     executionRoleArn: exampleAwsIamRole.arn,
 *     model: {
 *         bedrockModelConfig: {
 *             modelId: "anthropic.claude-sonnet-4-20250514",
 *             temperature: 0.7,
 *             topP: 0.9,
 *         },
 *     },
 *     systemPrompts: [{
 *         text: "You are a coding assistant.",
 *     }],
 *     allowedTools: ["*"],
 *     maxIterations: 10,
 *     maxTokens: 4096,
 *     timeoutSeconds: 300,
 *     tools: [{
 *         type: "inline_function",
 *         name: "get_weather",
 *         config: {
 *             inlineFunction: {
 *                 description: "Get the current weather for a location",
 *                 inputSchema: JSON.stringify({
 *                     type: "object",
 *                     properties: {
 *                         location: {
 *                             type: "string",
 *                             description: "City name",
 *                         },
 *                     },
 *                     required: ["location"],
 *                 }),
 *             },
 *         },
 *     }],
 *     truncations: [{
 *         strategy: "sliding_window",
 *         config: [{
 *             slidingWindow: [{
 *                 messagesCount: 50,
 *             }],
 *         }],
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * - `harnessId` (String) ID of the harness.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Bedrock AgentCore Harnesses using `harnessId`. For example:
 *
 * ```sh
 * $ pulumi import aws:bedrock/agentcoreHarness:AgentcoreHarness example example-Ab12Cd34Ef
 * ```
 */
class AgentcoreHarness extends pulumi.CustomResource {
    /**
     * Get an existing AgentcoreHarness resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AgentcoreHarness(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:bedrock/agentcoreHarness:AgentcoreHarness';
    /**
     * Returns true if the given object is an instance of AgentcoreHarness.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AgentcoreHarness.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowedTools"] = state?.allowedTools;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["authorizerConfiguration"] = state?.authorizerConfiguration;
            resourceInputs["environmentArtifact"] = state?.environmentArtifact;
            resourceInputs["environmentVariables"] = state?.environmentVariables;
            resourceInputs["environments"] = state?.environments;
            resourceInputs["executionRoleArn"] = state?.executionRoleArn;
            resourceInputs["harnessId"] = state?.harnessId;
            resourceInputs["harnessName"] = state?.harnessName;
            resourceInputs["maxIterations"] = state?.maxIterations;
            resourceInputs["maxTokens"] = state?.maxTokens;
            resourceInputs["memory"] = state?.memory;
            resourceInputs["model"] = state?.model;
            resourceInputs["region"] = state?.region;
            resourceInputs["skills"] = state?.skills;
            resourceInputs["systemPrompts"] = state?.systemPrompts;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeoutSeconds"] = state?.timeoutSeconds;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["tools"] = state?.tools;
            resourceInputs["truncations"] = state?.truncations;
        }
        else {
            const args = argsOrState;
            if (args?.executionRoleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'executionRoleArn'");
            }
            if (args?.harnessName === undefined && !opts.urn) {
                throw new Error("Missing required property 'harnessName'");
            }
            if (args?.model === undefined && !opts.urn) {
                throw new Error("Missing required property 'model'");
            }
            resourceInputs["allowedTools"] = args?.allowedTools;
            resourceInputs["authorizerConfiguration"] = args?.authorizerConfiguration;
            resourceInputs["environmentArtifact"] = args?.environmentArtifact;
            resourceInputs["environmentVariables"] = args?.environmentVariables ? pulumi.secret(args.environmentVariables) : undefined;
            resourceInputs["environments"] = args?.environments;
            resourceInputs["executionRoleArn"] = args?.executionRoleArn;
            resourceInputs["harnessName"] = args?.harnessName;
            resourceInputs["maxIterations"] = args?.maxIterations;
            resourceInputs["maxTokens"] = args?.maxTokens;
            resourceInputs["memory"] = args?.memory;
            resourceInputs["model"] = args?.model;
            resourceInputs["region"] = args?.region;
            resourceInputs["skills"] = args?.skills;
            resourceInputs["systemPrompts"] = args?.systemPrompts;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeoutSeconds"] = args?.timeoutSeconds;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["tools"] = args?.tools;
            resourceInputs["truncations"] = args?.truncations;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["harnessId"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["environmentVariables"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(AgentcoreHarness.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AgentcoreHarness = AgentcoreHarness;
