// @pulumi/aws/bedrock/agentcoreEvaluator.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentcoreEvaluator = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../utilities"));
/**
 * Manages an AWS Bedrock AgentCore Evaluator. An evaluator scores how an agent performs. You can configure it in one of two ways: an LLM-as-a-Judge evaluator that uses a model to score agent behavior against your instructions and a rating scale, or a code-based evaluator that runs a Lambda function you provide.
 *
 * ## Example Usage
 *
 * ### LLM-as-a-Judge with Numerical Rating Scale
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.bedrock.AgentcoreEvaluator("example", {
 *     evaluatorName: "helpfulness_evaluator",
 *     description: "Rates assistant helpfulness from 1 to 5",
 *     level: "TRACE",
 *     evaluatorConfig: {
 *         llmAsAJudge: {
 *             instructions: "Given the {context} and the {assistant_turn}, compare against {expected_response} and rate from 1 to 5.",
 *             ratingScale: {
 *                 numericals: [
 *                     {
 *                         definition: "Not helpful at all.",
 *                         value: 1,
 *                         label: "1",
 *                     },
 *                     {
 *                         definition: "Extremely helpful.",
 *                         value: 5,
 *                         label: "5",
 *                     },
 *                 ],
 *             },
 *             modelConfig: {
 *                 bedrockEvaluatorModelConfig: {
 *                     modelId: "us.amazon.nova-2-lite-v1:0",
 *                     inferenceConfig: {
 *                         maxTokens: 1024,
 *                         temperature: 0,
 *                         topP: 1,
 *                     },
 *                 },
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ### LLM-as-a-Judge with Categorical Rating Scale
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.bedrock.AgentcoreEvaluator("example", {
 *     evaluatorName: "tone_evaluator",
 *     level: "SESSION",
 *     evaluatorConfig: {
 *         llmAsAJudge: {
 *             instructions: "Classify the tone of the {assistant_turn} given the {context}.",
 *             ratingScale: {
 *                 categoricals: [
 *                     {
 *                         definition: "Friendly, helpful tone.",
 *                         label: "POSITIVE",
 *                     },
 *                     {
 *                         definition: "Neutral or terse tone.",
 *                         label: "NEUTRAL",
 *                     },
 *                     {
 *                         definition: "Unhelpful or dismissive tone.",
 *                         label: "NEGATIVE",
 *                     },
 *                 ],
 *             },
 *             modelConfig: {
 *                 bedrockEvaluatorModelConfig: {
 *                     modelId: "us.amazon.nova-2-lite-v1:0",
 *                 },
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ### Code-based Evaluator (Lambda)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.bedrock.AgentcoreEvaluator("example", {
 *     evaluatorName: "lambda_evaluator",
 *     level: "TOOL_CALL",
 *     evaluatorConfig: {
 *         codeBased: {
 *             lambdaConfig: {
 *                 lambdaArn: exampleAwsLambdaFunction.arn,
 *                 lambdaTimeoutInSeconds: 60,
 *             },
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `evaluatorId` (String) Unique identifier of the evaluator.
 *
 * #### Optional
 *
 * * `accountId` (String) AWS Account where this resource is managed.
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import Bedrock AgentCore Evaluator using the evaluator ID. For example:
 *
 * ```sh
 * $ pulumi import aws:bedrock/agentcoreEvaluator:AgentcoreEvaluator example helpfulness_evaluator-abc1234567
 * ```
 */
class AgentcoreEvaluator extends pulumi.CustomResource {
    /**
     * Get an existing AgentcoreEvaluator resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AgentcoreEvaluator(name, state, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'aws:bedrock/agentcoreEvaluator:AgentcoreEvaluator';
    /**
     * Returns true if the given object is an instance of AgentcoreEvaluator.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AgentcoreEvaluator.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["description"] = state?.description;
            resourceInputs["evaluatorArn"] = state?.evaluatorArn;
            resourceInputs["evaluatorConfig"] = state?.evaluatorConfig;
            resourceInputs["evaluatorId"] = state?.evaluatorId;
            resourceInputs["evaluatorName"] = state?.evaluatorName;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["level"] = state?.level;
            resourceInputs["lockedForModification"] = state?.lockedForModification;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.evaluatorConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'evaluatorConfig'");
            }
            if (args?.evaluatorName === undefined && !opts.urn) {
                throw new Error("Missing required property 'evaluatorName'");
            }
            if (args?.level === undefined && !opts.urn) {
                throw new Error("Missing required property 'level'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["evaluatorConfig"] = args?.evaluatorConfig;
            resourceInputs["evaluatorName"] = args?.evaluatorName;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["level"] = args?.level;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["evaluatorArn"] = undefined /*out*/;
            resourceInputs["evaluatorId"] = undefined /*out*/;
            resourceInputs["lockedForModification"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AgentcoreEvaluator.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AgentcoreEvaluator = AgentcoreEvaluator;
