// @pulumi/pulumiservice/api/esc/environmentSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentSchedule = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
/**
 * Creates a new scheduled action for a Pulumi ESC environment. Schedules can be used to automate recurring operations on environments, such as secret rotation. The request body specifies the schedule timing and the action to perform. Returns the created ScheduledAction on success. Requires the secret rotation feature to be enabled for the organization.
 */
class EnvironmentSchedule extends pulumi.CustomResource {
    /**
     * Get an existing EnvironmentSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnvironmentSchedule(name, undefined, { ...opts, id: id });
    }
    /** @internal */
    static __pulumiType = 'pulumiservice:api/esc:EnvironmentSchedule';
    /**
     * Returns true if the given object is an instance of EnvironmentSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentSchedule.__pulumiType;
    }
    /**
     * Create a EnvironmentSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if (args?.envName === undefined && !opts.urn) {
                throw new Error("Missing required property 'envName'");
            }
            if (args?.orgName === undefined && !opts.urn) {
                throw new Error("Missing required property 'orgName'");
            }
            if (args?.projectName === undefined && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            resourceInputs["envName"] = args?.envName;
            resourceInputs["orgName"] = args?.orgName;
            resourceInputs["projectName"] = args?.projectName;
            resourceInputs["scheduleCron"] = args?.scheduleCron;
            resourceInputs["scheduleID"] = args?.scheduleID;
            resourceInputs["scheduleOnce"] = args?.scheduleOnce;
            resourceInputs["secretRotationRequest"] = args?.secretRotationRequest ? pulumi.secret(args.secretRotationRequest) : undefined;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["definition"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastExecuted"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["nextExecution"] = undefined /*out*/;
            resourceInputs["orgID"] = undefined /*out*/;
            resourceInputs["paused"] = undefined /*out*/;
        }
        else {
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["definition"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastExecuted"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["nextExecution"] = undefined /*out*/;
            resourceInputs["orgID"] = undefined /*out*/;
            resourceInputs["paused"] = undefined /*out*/;
            resourceInputs["scheduleCron"] = undefined /*out*/;
            resourceInputs["scheduleID"] = undefined /*out*/;
            resourceInputs["scheduleOnce"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EnvironmentSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EnvironmentSchedule = EnvironmentSchedule;
