// @pulumi/pulumiservice/api/integrations/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitLabIntegration = exports.GitHubIntegration = exports.GitHubEnterpriseIntegration = exports.CustomVCSRepository = exports.CustomVCSIntegration = exports.BitBucketIntegration = exports.AzureDevOpsIntegration = void 0;
const pulumi = __importStar(require("@pulumi/pulumi"));
const utilities = __importStar(require("../../utilities"));
exports.AzureDevOpsIntegration = null;
utilities.lazyLoad(exports, ["AzureDevOpsIntegration"], () => require("./azureDevOpsIntegration"));
exports.BitBucketIntegration = null;
utilities.lazyLoad(exports, ["BitBucketIntegration"], () => require("./bitBucketIntegration"));
exports.CustomVCSIntegration = null;
utilities.lazyLoad(exports, ["CustomVCSIntegration"], () => require("./customVCSIntegration"));
exports.CustomVCSRepository = null;
utilities.lazyLoad(exports, ["CustomVCSRepository"], () => require("./customVCSRepository"));
exports.GitHubEnterpriseIntegration = null;
utilities.lazyLoad(exports, ["GitHubEnterpriseIntegration"], () => require("./gitHubEnterpriseIntegration"));
exports.GitHubIntegration = null;
utilities.lazyLoad(exports, ["GitHubIntegration"], () => require("./gitHubIntegration"));
exports.GitLabIntegration = null;
utilities.lazyLoad(exports, ["GitLabIntegration"], () => require("./gitLabIntegration"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "pulumiservice:api/integrations:AzureDevOpsIntegration":
                return new exports.AzureDevOpsIntegration(name, undefined, { urn });
            case "pulumiservice:api/integrations:BitBucketIntegration":
                return new exports.BitBucketIntegration(name, undefined, { urn });
            case "pulumiservice:api/integrations:CustomVCSIntegration":
                return new exports.CustomVCSIntegration(name, undefined, { urn });
            case "pulumiservice:api/integrations:CustomVCSRepository":
                return new exports.CustomVCSRepository(name, undefined, { urn });
            case "pulumiservice:api/integrations:GitHubEnterpriseIntegration":
                return new exports.GitHubEnterpriseIntegration(name, undefined, { urn });
            case "pulumiservice:api/integrations:GitHubIntegration":
                return new exports.GitHubIntegration(name, undefined, { urn });
            case "pulumiservice:api/integrations:GitLabIntegration":
                return new exports.GitLabIntegration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("pulumiservice", "api/integrations", _module);
